tox.ini

```diff
@@ -2,13 +2,22 @@
 envlist = py38-django{32,40}-celery{44,50}-drf{latest}
 
 [testenv]
+run_before = 
+    # Let tox control the versions for Django, DRF and celery for testing
+	sed -i.tmp '/^django==/d;/^vine==/d;/^amqp==/d;/^anyjson==/d' {toxinidir}/requirements/base.txt
+	sed -i.tmp '/^djangorestframework==/d;/^billiard==/d' {toxinidir}/requirements/base.txt
+	sed -i.tmp '/^celery==/d;/^kombu==/d' {toxinidir}/requirements/base.txt
+	rm {toxinidir}/requirements/base.txt.tmp
```

Comment:
This handles removing the constraints before running tox, but doesn't add them back afterwards.  It would probably be better to make these changes in a new copy of base.txt (tox_temp.txt?), use that copy in the deps section below, and then remove the temp file afterwards.  Should also add the temporary file to .gitignore in case tox gets interrupted and the file isn't properly removed.

So I think you'd want tox_temp.txt and test_layer.txt in the deps section instead of the test.txt convenience file, to avoid needing a temporary copy of that also (since it references base.txt explicitly).

Also, note that there's a bug in the existing code in that it never tests the exact versions of Celery, Django, etc. specified in the requirements files, it always goes for the latest release in the ranges specified here.  I think Ned fixed this in edx-platform, but the new setup may require a slightly different solution than what he did there.